### PR TITLE
[9.0][MIG] partner_firstname: Port to v9.0

### PR DESCRIPTION
**Pull request by kkarolis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ addon | version | summary
 --- | --- | ---
 [base_location](base_location/) | 9.0.1.0.0 | Enhanced zip/npa management system
 [partner_contact_nationality](partner_contact_nationality/) | 9.0.1.0.0 | Add nationality field to contacts
+[partner_firstname](partner_firstname/) | 9.0.2.0.0 | Split first name and last name for non company partners
```
Comment (elicoidal) [inline README.md:25]: hi @kkarolis Thanks for your contribution to the OCA. 
Please follow the guidelines from:
https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md

There should be no md file (for repo), only a README.rst, following: https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: partner_firstname/__openerp__.py [hunk 2]
```diff
@@ -21,7 +21,7 @@
 {
```
Comment (elicoidal) [inline partner_firstname/__openerp__.py:6]: Please use simple headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

File: partner_firstname/__openerp__.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
...
     'name': 'Partner first name and last name',
     'summary': "Split first name and last name for non company partners",
-    'version': '8.0.2.0.0',
+    'version': '9.0.2.0.0',
```
Comment (elicoidal) [inline partner_firstname/__openerp__.py]: 9.0.1.0.0

File: partner_firstname/__openerp__.py [hunk 4]
```diff
@@ -21,7 +21,7 @@
...
     'author': "Camptocamp,Odoo Community Association (OCA)",
     'maintainer': 'Camptocamp, Acsone',
```
Comment (elicoidal) [inline partner_firstname/__openerp__.py]: remove the deprecated key

File: partner_firstname/__openerp__.py [hunk 5]
```diff
@@ -35,6 +35,6 @@
     'demo': [],
```
Comment (elicoidal) [inline partner_firstname/__openerp__.py]: Remove empty keys

File: partner_firstname/models.py [hunk 6]
```diff
@@ -38,6 +38,11 @@ class ResPartner(models.Model):
         required=False,
         store=True)
 
+    # Disable sql name checking constraint
```
Comment (elicoidal) [inline partner_firstname/models.py:26]: Please use simple headers following OCA templates
https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3

**Comment by kkarolis** [comment]
Hey @elicoidal, thanks for your comments. I've updated the files to include new license headers.

**Comment by yvaucher** [comment]
Here we go 3 duplicates of this migration

Issue https://github.com/OCA/partner-contact/issues/184 was there to avoid this. :cry:

File: partner_firstname/__init__.py [hunk 7]
```diff
@@ -1,21 +1,6 @@
...
-#    License, or (at your option) any later version.
-#
-#    This program is distributed in the hope that it will be useful,
-#    but WITHOUT ANY WARRANTY; without even the implied warranty of
-#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-#    GNU Affero General Public License for more details.
-#
-#    You should have received a copy of the GNU Affero General Public License
-#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# © 2014 Agile Business Group (<http://www.agilebg.com>)
```
Comment (yvaucher) [inline partner_firstname/__init__.py:2]: I disagree as camptocamp copyright was removed

File: partner_firstname/views/res_partner.xml [hunk 8]
```diff
@@ -1,98 +1,85 @@
 <?xml version="1.0" encoding="utf-8"?>
-<openerp>
-<data>
+<odoo>
+    <data>
```
Comment (yvaucher) [inline partner_firstname/views/res_partner.xml:3]: data is non mandatory in odoo tag

File: partner_firstname/views/res_partner.xml [hunk 9]
```diff
@@ -1,98 +1,85 @@
...
-                    <field name="firstname" attrs=
-                        "{'required': [('lastname', '=', False),
-                                       ('is_company', '=', False)]}"/>
-                </group>
-            </xpath>
+                    <!-- Modify inner contact form of child_ids -->
+                    <xpath expr="//field[@name='child_ids']/form
+                        //field[@name='name']"
+                        position="before">
+                        <field name="is_company"/>
```
Comment (yvaucher) [inline partner_firstname/views/res_partner.xml:62]: Can a contact be a company? If we want to display it, we may want to display "company_type" field as in main form view

**Comment by yvaucher** [comment]
@kkarolis I close this PR in favor of https://github.com/OCA/partner-contact/pull/238

Which takes history from 8.0 branch and added missing authors, plus is adapted for shipping and invoicing addresses which can have no name in 9.0 done by @alnslang in #232 

I hope you don't mind I tried to regroup the changes of all duplicate PRs. If I'm missing something please tell me.

**Pull request closed without merging**